src/spec.c

```diff
@@ -1379,33 +1349,28 @@ void IndexSpec_Free(IndexSpec *spec) {
   // next time it will try to acquire the spec.
 
   // For temporary index
-  if (spec->isTimerSet) {
-    WeakRef old_timer_ref;
-    if (RedisModule_StopTimer(RSDummyContext, spec->timerId, (void **)&old_timer_ref) == REDISMODULE_OK) {
-      WeakRef_Release(old_timer_ref);
-    }
-    spec->isTimerSet = false;
-  }
+  // This function might be called from any thread, and we cannot deal with timers without the GIL.
+  // At this point we should have already stopped the timer.
+  assert(!spec->isTimerSet);
   // Stop and destroy indexer
   if (spec->indexer) {
     Indexer_Free(spec->indexer);
   }
   // Stop and destroy garbage collector
-  if (spec->gc) {
-    GCContext_Stop(spec->gc);
+  // We can't free it now, because it eighter runs at the moment or has a timer set which we can't
+  // deal with without the GIL.
+  // It will free itself when it discovers that the index was freed.
+  // On the worst case, it just finishes the current run and will schedule another run soon.
+  // In this case the GC will be freed on the next run, in `forkGcRunIntervalSec` seconds.
+  if (RS_IsMock && spec->gc) {
```

Comment:
I do not like it that we have so many code branching just for testing. What does the mock does and what does the test check? Maybe we can avoid it?

Reply:
I have no idea. I’ll try to check. I see that this code is covered.
Maybe we should add a compilation flag for unit tests and just have this lines compiled-out for regular compilation?

Reply:
Anyway this is a destructor function. Not in use too often

Reply:
Lets keep it for the scope of this PR, I think we should delete this entire mock if it caused to much branches from the production code ...